lib/src/component_declaration/component_base.dart

```diff
@@ -809,3 +815,39 @@ class ConsumedProps {
 
   const ConsumedProps(this.props, this.keys);
 }
+
+abstract class AccessorMeta<T extends _Descriptor> {
+  List<T> get fields;
+  List<String> get keys;
+}
+
+class PropsMeta implements ConsumedProps, AccessorMeta<PropDescriptor> {
```

Comment:
Need to add doc comments for this class and `StateMeta`.. but I'm honestly not sure how we should describe these classes. @greglittlefield-wf could I get an assist from you?

Reply:
How about something like this?
```dart
/// Metadata for the prop fields declared in a specific props class--
/// a class annotated with @[Props], @[PropsMixin], @[AbstractProps], etc. 
/// for which prop accessors are generated.
///
/// This metadata includes map key values corresponding to these fields,
/// which is used in [UiComponent.consumedPropKeys],
/// as well as other prop configuration done via @[Accessor]/@[requiredProp]/etc.,
/// which is used to perform prop validation within [UiComponent] lifecycle methods.
///
/// This metadata is generated as part of the over_react builder, and should
/// be exposed like so:
///     @Props()
///     class FooProps {
///       static const PropsMeta meta = $metaForFooProps;
///     
///       String foo;
///
///       @Accessor(isRequired: true, key: 'custom_key', keyNamespace: 'custom_namespace')
///       int bar;
///     }
///
/// What the metadata looks like:
///     main() {
///       print(FooProps.meta.keys); // [FooProps.foo, custom_namespace.custom_key]
///       print(FooProps.meta.props.map((p) => p.isRequired); // (false, true))
///     }
///
/// _See also: [getPropKey]_
```